File: Stacks-and-Queues/stack.rb
module Stack

class Node
  attr_accessor :value, :next
  def initialize(value)
    @value = value 
    @next = nil
  end
end

class Stack
  attr_accessor :top, :length
  def initialize(value)
    new_node = Node.new(value)
    @top = new_node
    @length = 1
  end

  def push(value)
    new_node = Node.new(value)
    if @top == nil
      @top = new_node
    else
      new_node.next = @top
      @top = new_node
    end
    @length += 1
    return self 
  end
  
  def pop()
    if @top == nil
      return nil
    end
    temp = @top
    @top = @top.next
    temp.next = nil
    @length -= 1
    return temp
  end
end
end

File: Stacks-and-Queues/stack.py
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None


class Stack:
    def __init__(self, value):
        new_node = Node(value)
        self.top = new_node
        self.length = 1

    def push(self, value):
        new_node = Node(value);
        if self.top == None:
            self.top = new_node
        else:
            new_node.next = self.top
            self.top = new_node

        self.length += 1
        return self

    def pop(self): 
        if self.top == None:
            return None
        temp = self.top
        self.top = self.top.next
        temp.next = None
        self.length -= 1
        return temp


